### PR TITLE
Replace site restricted token implementation with SWT

### PR DESCRIPTION
**Pull request by ahmelsayed**

### CODE REVIEW

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 1]
```diff
@@ -90,16 +90,16 @@ public static bool IsAzureEnvironment()
         /// It is written to require least dependencies but framework assemblies.
         /// Caller is responsible for synchronization.
         /// </summary>
-        public static async Task Run(string requestId, string siteRestrictedJwt, TraceListener tracer)
```
[the remark this answers is not in the document]
Comment (ahmelsayed) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: I just added `siteRestrictedJwt` and added a suppress rule for fxcop error about unused parameter

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 2]
```diff
@@ -417,7 +417,7 @@ private static void WriteAutoSwapOngoing()
                 {
                     client.BaseAddress = new Uri(string.Format("https://{0}", host));
                     client.DefaultRequestHeaders.UserAgent.Add(_userAgent.Value);
-                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", siteRestrictedJwt);
+                    client.DefaultRequestHeaders.Add(Constants.SiteRestrictedToken, SimpleWebTokenHelper.CreateToken(DateTime.UtcNow.AddMinutes(2)));
```
[the remark this answers is not in the document]
Comment (ahmelsayed) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: fixed

File: Kudu.Core/Helpers/SimpleWebTokenHelper.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        /// <summary>
+        /// A SWT or a Simple Web Token is a token that's made of key=value pairs seperated
+        /// by &. We only specify expiration in ticks from now (exp={ticks})
+        /// The SWT is then returned as an encrypted string
+        /// </summary>
+        /// <param name="validUntil">Datetime for when the token should expire</param>
+        /// <returns>a SWT signed by this app</returns>
+        public static string CreateToken(DateTime validUntil) => Encrypt($"exp={validUntil.Ticks}");
+
+
```
[the remark this answers is not in the document]
Comment (ahmelsayed) [reply Kudu.Core/Helpers/SimpleWebTokenHelper.cs]: fixed

File: Kudu.Core/Helpers/SimpleWebTokenHelper.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        /// A SWT or a Simple Web Token is a token that's made of key=value pairs seperated
+        /// by &. We only specify expiration in ticks from now (exp={ticks})
+        /// The SWT is then returned as an encrypted string
+        /// </summary>
+        /// <param name="validUntil">Datetime for when the token should expire</param>
+        /// <returns>a SWT signed by this app</returns>
+        public static string CreateToken(DateTime validUntil) => Encrypt($"exp={validUntil.Ticks}");
+
+        [SuppressMessage("Microsoft.Usage", "CA2202:Object 'cipherStream' and 'cryptoStream' can be disposed mo re than once",
+            Justification = "MemoeryStream, CryptoStream, and BinaryWriter handle multiple disposal correctly. The alternative is pretty ugly code for clearing each variable, checking for null, and manual dispose.")]
```
Comment (ahmelsayed) [inline Kudu.Core/Helpers/SimpleWebTokenHelper.cs]: CA2202 is stupid

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 5]
```diff
@@ -91,16 +92,18 @@ public static bool IsAzureEnvironment()
         /// It is written to require least dependencies but framework assemblies.
         /// Caller is responsible for synchronization.
         /// </summary>
+        [SuppressMessage("Microsoft.Usage", "CA1801:Parameter 'siteRestrictedJwt' is never used", 
+            Justification = "Method signature has to be the same because it's called via reflections from web-deploy")]
         public static async Task Run(string requestId, string siteRestrictedJwt, TraceListener tracer)
```
Comment (suwatch) [inline Kudu.Core/Helpers/PostDeploymentHelper.cs]: I wonder if we could introduce Run2(string requestId, TraceListener tracer) and have this Run calling it.   Once we do that, MSDeploy can switch to call Run2 and we can get rid of the Run.    Open for better name than Run2.

Comment (ahmelsayed) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: how about `RunPostDeployment()` or `Invoke()`? any preference?

Comment (ahmelsayed) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: or `Run2()` if you don't like either one

Comment (ahmelsayed) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: `Invoke()` it is

**Pull request merged**